docs/NXP_PXP_readme.md

```diff
@@ -0,0 +1,59 @@
+
+# HW acceleration for NXP iMX RT platforms using PXP (PiXel Pipeline) engine 
+
+Several drawing features in LVGL can be offloaded to PXP engine. In order to use CPU time while PXP is running, RTOS is required to block LVGL drawing thread and switch to another task, or simply to idle task, where CPU could be suspended to save power. 
+
+
+#### Features supported:
+- RGB565 format only
+- Area fill + optional transparency
+- BLIT (image transfer) + optional transparency
+- Color keying + optional transparency
+- Recoloring (color tint) + optional transparency
+- RTOS integration layer
+- default FreeRTOS and bare metal code provided
+
+#### Basic configuration:
+  - enable GPU support in lv_conf.h: Set LV_USE_GPU to 1
+  - select NXP PXP engine: Set LV_USE_GPU_NXP_PXP to 1 
+  - enable default implementation for interrupt handling and PXP start function: Set LV_USE_GPU_NXP_PXP_DEFAULT_OSA to 1
+    - if FSL_RTOS_FREE_RTOS symbol is defined, FreeRTOS implementation will be selected, otherwise bare metal code will be 
+      included. 
+    
+#### Basic initialization: 
+  - Initialize PXP module before LVGL library. Default configuration structure for callbacks can be used: 
+ 
+```c
+      #if LV_USE_GPU && LV_USE_GPU_NXP_PXP
```

Comment:
We initialize the DMA2D in [lv_init()](https://github.com/lvgl/lvgl/blob/e3a07ed58cc53bbe6e259dd16a5976aaade5cab9/src/lv_core/lv_obj.c#L153). Can we do the same with PXP?

Reply:
I chosed this approach to give the user a chance to do the OS specific low-level handling on his own by providing callback functions. If we move pxp init into `lv_init()`, we would either force user to use default implementation of callbacks (baremetal and FreeRTOS), or to change LVLG code, so init will be called with user specified callbacks. Possible case is when user integrate LVGL + PXP to RTOS other than FreeRTOS. In this case new callbacks has to be provided, so user will need to alter lv_init() function.

Reply:
Finally pxp init needs an `lv_nxp_pxp_cfg_t` structure which can be the default or user-specified, right?

If so  could it work in `lv_init()`?
```c
#if LV_USE_GPU_NXP_PXP_DEFAULT_OSA
        if (lv_gpu_nxp_pxp_init(&pxp_default_cfg) != LV_RES_OK) {
#else 
        extern lv_nxp_pxp_cfg_t pxp_custom_cfg;       //hard coded name, user must use this
        if (lv_gpu_nxp_pxp_init(&pxp_custom_cfg) != LV_RES_OK) {
#endif 
```

I'm asking this to reduce the tasks that need to be done by the user (to reduce the number of possible issues).

Reply:
I understand, good point. However we need to add comment to that ``LV_USE_GPU_NXP_PXP_DEFAULT_OSA`` variable, that only FreeRTOS is detected via ``FSL_RTOS_FREE_RTOS`` symbol.  Because currently there's construction like:
```
#if defined(FSL_RTOS_FREE_RTOS)
#else
#endif
```
So if user try to integrate into different RTOS and have enabled ``LV_USE_GPU_NXP_PXP_DEFAULT_OSA``, it will silently go to bare-metal branch, which is wrong. 

I am not sure about the #else branch - pxp configuration is copied internally into static variable in [``lv_gpu_nxp_pxp_init``](https://github.com/NXPmicro/lvgl/blob/a01b7a448bd1a554d70e0c278e47e7c1c60962c6/src/lv_gpu/lv_gpu_nxp_pxp.c#L108). If there is extern pxp_custom_cfg, it gets duplicated. Moreover, this makes "secret" dependency on exact variable name and brings new symbol into global namespace. 

What about renaming LV_USE_GPU_NXP_PXP_DEFAULT_OSA to LV_USE_GPU_NXP_PXP_AUTO_INIT ? That may be more clear. Add it to lv_init. Otherwise let user to init pxp on his own, so we avoid issues mentioned above.

Reply:
> What about renaming LV_USE_GPU_NXP_PXP_DEFAULT_OSA to LV_USE_GPU_NXP_PXP_AUTO_INIT ? That may be more clear. Add it to lv_init. Otherwise let user to init pxp on his own, so we avoid issues mentioned above.

Perfect! It's really clearer.

Reply:
Great, I'll rename the symbol, add default init into lv_init, update docs and add new patch to PR.

Reply:
I have added new commit, but auto-test failed with error I don't understand. Could you have a look please and help me to resolve problem? How can I run that micro-python test locally before push?

Reply:
The MicroPython test is not working properly yet; I don't see why your changes would cause any issues though.

Reply:
So is that a test setup problem? Is it allowed to approve PR even when auto-test is failing?

Reply:
Hello @kisvegabor , @embeddedt , do you have any idea please how can I fix the auto-build error?

Reply:
> do you have any idea please how can I fix the auto-build error?

Don't worry about it; it needs to be fixed on our end.